### PR TITLE
Adding support for alternate authorization bindings (#4)

### LINKED ISSUE #4: MetadataReader - Add Support For HTTP-POST/HTTP-Artifact Authorization Bindings
At present the MetadataReader only supports `HTTP-Redirect` authorization bindings.  Would be good to add a new option for `authnRequestBinding` to set a preferred binding in the reader (variable name matches [passport-saml](https://github.com/bergie/passport-saml#config-parameter-details)); and to fallback to one of the other two bindings if the specified/default binding is not present.

### PR DESCRIPTION
**Pull request by TigerC10**
Resolves #4

* Adds a new option `authnRequestBinding` to specify the binding to use for identityProviderUrl/logoutUrl
* `authnRequestBinding` defaults to `'HTTP-Redirect'`
* If specified `authnRequestBinding` is not present in metadata, defaults to the highest priority binding instead

### CODE REVIEW

**Comment by TigerC10** [comment]
Fetch test cases are failing because I changed the metadata so that there could be unit tests for the alternate bindings.

File: README.md [hunk 1]
```diff
@@ -100,6 +100,10 @@ Parses metadata XML and extracts the following properties:
 * `encryptionCert`
 * `claimSchema` - an object hash of claim identifiers that may be provided in the SAML assertion
 
+#### Options parameter details:
```
Comment (compwright) [inline README.md]: Could you remove this heading and make the options all one list?

Comment (TigerC10) [reply README.md]: The bulleted list above is the structure of the return, this heading separates the return structure from the structure of the options argument.  Don't think we should combine the bulleted lists here, but happy to re-organize (maybe position the options above the return?)

Comment (compwright) [reply README.md]: Oh okay, I follow. Perhaps we should move the list of options ahead of the list of return values? What do you think? This just struck me as being a little confusing.

Comment (TigerC10) [reply README.md]: Yeah, it felt awkward to me both above and below.  Will relocate it though;  Might be more helpful to use jsdoc block comments above the functions to document the options better.  Could then take that generated docset and publish to GitHub pages for better clarity.

**Comment by compwright** [comment]
Published as v1.5.0. Thanks again!

**Comment by TigerC10** [comment]
No, thank you! 😸 So glad this project exists, happy to contribute back to it.

**Pull request merged**